src/phoenix/experimental/evals/models/openai.py

```diff
@@ -135,6 +176,18 @@ def _init_tiktoken(self) -> None:
             encoding = self._tiktoken.get_encoding("cl100k_base")
         self._tiktoken_encoding = encoding
 
+    def _get_azure_options(self) -> AzureOptions:
+        options = {}
+        for option in AZURE_REQUIRED_OPTIONS:
+            value = getattr(self, option)
+            if value is None:
+                raise ValueError(f"Option '{option}' must be set when using Azure OpenAI API")
+            options[option] = value
+        for option in AZURE_ADDITIONAL_OPTIONS:
+            value = getattr(self, option)
+            options[option] = value
```

Comment:
fyi. here's a fancy way to do this. 
>from dataclasses import fields
>from typing import get_args, get_origin
```suggestion
        for option in fields(AzureOptions):
            if (value := getattr(self, option.name)) is not None:
                options[option.name] = value
                continue
            # raise ValueError if field is not optional
            if not (get_origin(option.type) is Union and type(None) in get_args(option.type)):
                raise ValueError(f"Option '{option.name}' must be set when using Azure OpenAI API")
```

Reply:
Oh yeah that is quite "fancy"... I'm going with it! But yeah first time using typing functions... feels advanced for me. haha